### ISSUE TITLE
Error when retrieving data area restricted by bounding box 

### ISSUE THREAD

**Issue by justus-saul** [HTenkanen/pyrosm#54]
Hi,
I was trying to extract some data in a bounding box with a Polgyon, but ran into an issue. You can see the traceback below.
I also duplicated the code from the docs and used shapely geometry from GeoDataFrame directly from an existing boundary in a pbf file, but got the same results. As I used the code directly from the docs, I have to assume that this is a bug in the library.

**Traceback**
This shows the traceback for getting pois, but it looks the same when getting other information (like in the docs code).
```
Traceback (most recent call last):
  File "/home/XXX/sandbox/osm/main.py", line 62, in <module>
    run_osm()
  File "/home/XXX/sandbox/osm/main.py", line 57, in run_osm
    pois = osm.get_pois(custom_filter=custom_filter)
  File "/home/XXX/sandbox/osm/venv/lib/python3.8/site-packages/pyrosm/pyrosm.py", line 492, in get_pois
    self._read_pbf()
  File "/home/XXX/sandbox/osm/venv/lib/python3.8/site-packages/pyrosm/pyrosm.py", line 85, in _read_pbf
    nodes, ways, relations, way_tags = parse_osm_data(self.filepath,
  File "pyrosm/pbfreader.pyx", line 310, in pyrosm.pbfreader.parse_osm_data
  File "pyrosm/pbfreader.pyx", line 312, in pyrosm.pbfreader.parse_osm_data
  File "pyrosm/pbfreader.pyx", line 327, in pyrosm.pbfreader._parse_osm_data
  File "pyrosm/pbfreader.pyx", line 124, in pyrosm.pbfreader.parse_dense
IndexError: boolean index did not match indexed array along dimension 0; dimension is 0 but corresponding boolean dimension is 3200446
```

Using **Python 3.8.5** on Manjaro.
See all dependencies below:
```
 • Installing click (7.1.2)
  • Installing attrs (20.2.0)
  • Installing click-plugins (1.1.1)
  • Installing cligj (0.5.0)
  • Installing markupsafe (1.1.1)
  • Installing munch (2.5.0)
  • Installing numpy (1.19.2)
  • Installing python-dateutil (2.8.1)
  • Installing pytz (2020.1)
  • Installing cython (0.29.21)
  • Installing fiona (1.8.17)
  • Installing jinja2 (2.11.2)
  • Installing pandas (1.1.3)
  • Installing pyproj (2.6.1.post1)
  • Installing shapely (1.7.1)
  • Installing cykhash (1.0.2)
  • Installing geopandas (0.8.1)
  • Installing pygeos (0.8)
  • Installing pyrobuf (0.9.3)
  • Installing python-rapidjson (0.9.1)
  • Installing pyrosm (0.5.3)
```

Code snippet:

``` python
    fp = get_data("Berlin")
    osm = OSM(fp)
    bounding_box = osm.get_boundaries(name="Friedrichshain")
    bbox_geom = bounding_box['geometry'].values[0]
    print(bbox_geom)
    osm = OSM(fp, bounding_box=bbox_geom)
    custom_filter = {'amenity': ["bar"]}
    pois = osm.get_pois(custom_filter=custom_filter)
    pois.to_csv("export.csv", sep='\t')
```

**Comment by HTenkanen** [comment]
@justus-saul Thanks for reporting this! I was able to reproduce the error and will take a look at this next week. 👍

**Comment by HTenkanen** [comment]
Hi @justus-saul! 

Sorry for taking rather long to get back to this. Just letting you know that this has been fixed and will be available in the next release (v0.6.0) that should be out soon. So following works as should now:

![image](https://user-images.githubusercontent.com/6659534/98705717-ac67b680-2386-11eb-80a5-f391bd0d4f8d.png)
